src/components/projects-section/index.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import "./style.css";
+import { useState } from "react";
+import { ProjectsIcon } from "./projects-icon";
+import { mock_projects_data, mock_projects_buttons } from "@/mock-data/projects";
+import { ProjectsCarousel } from "../projects-carousel";
+
+export function ProjectsSection() {
+  const [activeProjectIndex, setActiveProjectIndex] = useState(0);
+  const [direction, setDirection] = useState<"left" | "right">("right");
+
+  const handlePreviousSlide = () => {
+    setDirection("left");
+    if (activeProjectIndex === 0) return setActiveProjectIndex(mock_projects_data.length - 1);
+
+    return setActiveProjectIndex((prev) => prev - 1);
+  };
+
+  const handleNextSlide = () => {
+    setDirection("right");
+    if (activeProjectIndex === mock_projects_data.length - 1) {
+      return setActiveProjectIndex(0);
+    }
+
+    return setActiveProjectIndex((prev) => prev + 1);
+  };
```

Comment:
Se a ideia é mover o index atual para o próximo ou anterior, então acho que um nome mais adequado para o parâmetro aqui seja `curr`, porque você está calculando com base no _current_.
Com o nome `prev`, me dá a ideia de que esse estado é alterado sempre com base no anterior.
```suggestion
  const handlePreviousSlide = () => {
    setDirection("left");
    if (activeProjectIndex === 0) return setActiveProjectIndex(mock_projects_data.length - 1);

    return setActiveProjectIndex((curr) => curr - 1);
  };

  const handleNextSlide = () => {
    setDirection("right");
    if (activeProjectIndex === mock_projects_data.length - 1) {
      return setActiveProjectIndex(0);
    }

    return setActiveProjectIndex((curr) => curr + 1);
  };
```

Reply:
O uso de prev ou prevState é meio que uma convenção quando se usa uma função pra atualizar o estado no React (https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state). o parâmetro pode ter qualquer nome, mas por convenção, uma das possibilidades é usar o estado anterior pra calcular o próximo sem que ele calculado com base em um estado desatualizado.

Reply:
boa!